core/v2/v2.go

```diff
@@ -1077,11 +1075,6 @@ func (l *signatureTreeDynamicSignatureLeaf) recover(ctx context.Context, subdige
 			if magicValue != isValidSignatureMagicValue {
 				return nil, nil, fmt.Errorf("isValidSignature returned %v, expected %v", hexutil.Encode(magicValue[:]), hexutil.Encode(isValidSignatureMagicValue[:]))
 			}
-		} else {
```

Comment:
It's not present on master. It also makes Guard fail if we have nested EIP1271 signatures.

Reply:
Hmm this is wrong, we shouldn't remove this. If we don't have a provider we can't just assume the signature is valid, if the guard fails with nested signatures it means that the guard is not validating them while passing a provider.

This change is equivalent to ignoring nested signatures are assuming they are valid. Please revert it.

Keep in mind prod is not using `master`, it's using the `v2` branch.

Reply:
@Agusx1211 
The problem is that in the guard in not_first filter we are getting incorrect weight because of that. Since that filter requires signature weight of at least 1 it causes guard to reject sign request.

There are couple of questions to answer here:
1. Should Recover be equal to validation?
2. Should Recover return incorrect weight if signature can not be fully verified?
3. How do we differentiate between incomplete signature and not being able to verify? 
    - Only from call context? 
    - How do we know signature unverified weight then?

Reply:
This is happening because `not_first_signer` is built for wallet-webapp, and it's not expected to validate below threshold nested signatures. If WaaS doesn't need this filter, the proper solution is to disable the filter.

1. No, you can recover any signature, validation means `imageHash === wallet.imageHash` and `weight > threshold`. If a signature can't be verified (because we can't access the provider) then we must assume this signature is invalid.

2. If you don't provide a provider we have 3 options:
a) throwing an error
b) assuming nested signatures are valid
c) assuming nested signatures are invalid

security wise (a) and (c) are equivalent, (b) is a no-go because it opens the door for messing up when manually validating a signature.

3. because you don't have a provider, if you have the provider, then the state is no longer undetermined. `How do we know signature unverified weight then?` this question is backwards, an unverified signature doesn't have a weight, because you don't know if the inner signers will approve the signature or not

Reply:
Understood. All of it makes sense.

So I need to decide to either create yet another Guard deployment or add providers to the Guard itself.

I think I will just add providers. This seems much easier and faster to be done.